jetbrains-core/src/software/aws/toolkits/jetbrains/core/tools/SemanticVersion.kt

```diff
@@ -0,0 +1,31 @@
+// Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.core.tools
+
+data class SemanticVersion(val major: Int, val minor: Int, val patch: Int) : Version {
+    override fun displayValue(): String = "$major.$minor.$patch"
+
+    override fun compareTo(other: Version): Int = COMPARATOR.compare(this, other as SemanticVersion)
+
+    companion object {
+        // TODO: Support pre-release
+        private val COMPARATOR = compareBy<SemanticVersion> { it.major }
+            .thenBy { it.minor }
+            .thenBy { it.patch }
+
+        fun parse(version: String): SemanticVersion {
+            val parts = version.split(".")
+            if (parts.size != 3) {
+                throw IllegalArgumentException("$version not in the format of MAJOR.MINOR.PATCH")
+            }
+
+            try {
+                val patch = parts[2].substringBefore("-").toInt()
```

Comment:
prly also need to substring before `+` too while we're at it, this is a valid semantic version: `1.2.2+12423`

Reply:
Fair, there is a TODO a few lines up for pre-release comparison but we should def do it for parse I think

Reply:
Spec says its hyphen. (https://semver.org/#spec-item-9)

Reply:
Added